src/GafferScene/SetFilter.cpp

```diff
@@ -129,7 +169,12 @@ unsigned SetFilter::computeMatch( const ScenePlug *scene, const Gaffer::Context
 	}
 
 	const ScenePlug::ScenePath &path = context->get<ScenePlug::ScenePath>( ScenePlug::scenePathContextName );
-	ConstPathMatcherDataPtr set = scene->set( setPlug()->getValue() );
+
+	Gaffer::ContextPtr pathMatcherContext = new Gaffer::Context( *context, Gaffer::Context::Borrowed );
```

Comment:
Let's use the fancy new ScenePlug::GlobalScope here.

We've also got a mismatch between the `computeMatch()` and `hashMatch()` methods now - we need to hash in `expressionResultPlug()` in `hashMatch()`, rather than calling `setHash()`. We'll need to use a GlobalScope for that too.